content/artifacts/artifacts-finder/plugin.js

```diff
@@ -56,54 +60,113 @@ function createDom(tag, text) {
   return dom;
 }
 
+function sleep(ms) {
```

Comment:
This is .. not great. Theres definitely a way to wait for tx, but it is complicated and were pushing for a refactor so probably fine for now

Reply:
Yes, sleep is not good. I'll try to find a new solution to do this.